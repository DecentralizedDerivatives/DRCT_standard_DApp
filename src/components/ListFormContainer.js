import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { reduxForm, getFormValues } from 'redux-form';
import { sendListOrder } from '../actions/orderActions';
import ListFormComponent from './ListFormComponent';

const validate = values => {
  const errors = {};
  if (!values.token) {
    errors.token = 'Required';
  }

  if (!values.price) {
    errors.price = 'Required';
  }

  if (!values.tokenAmount) {
    errors.tokenAmount = 'Required';
  }

  return errors;
};

export let ListFormContainer = props => {
  const submitForm = (formValues, sendListOrder, userAccount) => {
    console.log('submitting Form: ', formValues);
    sendListOrder(formValues, userAccount);
  };

  return (
    <ListFormComponent
      formValues={props.formValues}
      change={props.change}
      onSubmit={submitForm}
      handleSubmit={props.handleSubmit}
      selectOptions={props.userTokens}
    />
  );
};

ListFormContainer.propTypes = {
  sendListOrder: PropTypes.func.isRequired,
  formValues: PropTypes.object,
  userTokens: PropTypes.array
};

const mapStateToProps = state => ({
  formValues: getFormValues('list-form')(state),
  userTokens: state.user.userTokens
});

const formConfiguration = {
  form: 'list-form',
  validate
};

ListFormContainer = reduxForm(formConfiguration)(ListFormContainer);

export default (ListFormContainer = connect(
  mapStateToProps,
  { sendListOrder }
)(ListFormContainer));
