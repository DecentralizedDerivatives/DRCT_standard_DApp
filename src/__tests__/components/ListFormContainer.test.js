import ListFormContainer from '../../components/ListFormContainer';

describe('<ListFormContainer />', () => {
  describe('render()', () => {
    it('renders the component', () => {
      const wrapper = shallow(
        <ListFormContainer store={initFixtureStore()} />
      ).dive().dive().dive().dive();

      expect(wrapper).toMatchSnapshot();
    });
  });
});
