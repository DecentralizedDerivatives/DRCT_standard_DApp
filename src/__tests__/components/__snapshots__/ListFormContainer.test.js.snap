// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListFormContainer /> render() renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListFormContainer
    anyTouched={false}
    array={
      Object {
        "insert": [Function],
        "move": [Function],
        "pop": [Function],
        "push": [Function],
        "remove": [Function],
        "removeAll": [Function],
        "shift": [Function],
        "splice": [Function],
        "swap": [Function],
        "unshift": [Function],
      }
    }
    asyncValidate={[Function]}
    asyncValidating={false}
    autofill={[Function]}
    blur={[Function]}
    change={[Function]}
    clearAsyncError={[Function]}
    clearFields={[Function]}
    clearSubmit={[Function]}
    clearSubmitErrors={[Function]}
    destroy={[Function]}
    dirty={false}
    dispatch={[Function]}
    form="list-form"
    handleSubmit={[Function]}
    initialize={[Function]}
    initialized={false}
    invalid={false}
    pristine={true}
    pure={true}
    reset={[Function]}
    resetSection={[Function]}
    sendListOrder={[Function]}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "unsubscribe": [Function],
        },
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
    submit={[Function]}
    submitFailed={false}
    submitSucceeded={false}
    submitting={false}
    token=""
    tokenAmount=""
    touch={[Function]}
    untouch={[Function]}
    userAccount="0xd8e09fbab17f30e1C2b2Ae85A26EA1F3F31267ef"
    valid={true}
    validate={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "handleSubmit": [Function],
      "onSubmit": [Function],
      "token": "",
      "tokenAmount": "",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "handleSubmit": [Function],
        "onSubmit": [Function],
        "token": "",
        "tokenAmount": "",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
